docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,274 @@
+# MeshExternalService and ZoneEgress
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/10897
+
+## Context and Problem Statement
+
+MeshExternalService traffic can go out directly from the sidecar or through egress.
+
+Configuration (like timeouts, retries, circuit breakers) of MeshExternalService can also be applied both on a sidecar and on an egress.
+
+This MADR will go into the details of:
+- limitations / features that occur / are possible when running the traffic directly from sidecar vs through egress
+- each policy type and the placement to figure out what is the best option
+
+## Decision Drivers
+
+* not restricting any useful use-cases
+* flexible enough that we don't limit users in scenarios where it makes sense to configuration in both/one place
+* restrictive enough so that it's hard for the user to "shoot themselves in the foot" (e.g. having squared no. of circuit breakers)
+
+## Considered Options
+
+Running traffic through:
+* egress only
+* sidecar only
+* mix of sidecar and egress
+
+Configuration placement:
+* configurable where it makes sense on both sidecar and egress, predefined where only one makes sense
+* predefined mixed approach on a case-by-case basis
+* only on the egress
+* only on the sidecar
+
+## Decision Outcome
+
+Chosen option: ?
+
+Chosen option for configuration placement: ?
+
+## Pros and Cons of the Options
+
+### Running traffic through egress only
+
+- there is a bit of operational cost to running egress
+- it's a bit tricky to set up on universal
+- we've got a bit of a gap in other functionalities on egress (configuration, observability)
+- mTLS is required to run egress
+- there is additional network hop
+- with the introduction of MeshPassthrough the barrier for external traffic is lower and some cases can be covered by that
+
+Conclusion: **it seems like we're not cutting any serious use case**.
+
+### Running traffic through sidecar only
+
+- all the problems of running through egress are not applicable (operational cost, extra networking hop, scalability etc.)
+- no single exit point for all the external traffic
+- if we put credentials in the ExternalService then these credentials are accessible to the app owners (need to check Envoy secrets)
```

Comment:
after the meeting I checked and I think we can just use `generic_secret` for this

```suggestion
- if we put credentials in the ExternalService then these credentials are accessible to the app owners if not delivered via generic_secret
```